elasticgraph-graphql/spec/integration/elastic_graph/graphql/datastore_query/filtering_spec.rb

```diff
@@ -150,6 +150,138 @@ class GraphQL
+              {
+                "past_names" => {
+                  "any_satisfy" => {"equal_to_any_of" => ["foo"]}
+                }
+              },
+              {
+                "forbes_valuations" => {
+                  "any_satisfy" => {"gt" => 100_000}
+                }
+              }
+            ]
+          }
+
+          results_all_of = search_with_freeform_filter(filter_all_of)
+          expect(results_all_of).to match_array(ids_of(t1))
+        end
+
+        it "handles AND of OR subfilters (allOf of anyOf)" do
+          index_into(
+            graphql,
+            build(:team, id: "t1", past_names: ["foo"], forbes_valuations: [500_000]),
+            t2 = build(:team, id: "t2", past_names: ["foo"], forbes_valuations: [50_000]),
+            t3 = build(:team, id: "t3", past_names: ["bar"], forbes_valuations: [500_000]),
+            build(:team, id: "t4", past_names: ["bar"], forbes_valuations: [50_000])
+          )
+
+          filter = {
+            "all_of" => [
+              {
+                "any_of" => [
+                  {"past_names" => {"any_satisfy" => {"equal_to_any_of" => ["foo"]}}},
+                  {"forbes_valuations" => {"any_satisfy" => {"gt" => 100_000}}}
+                ]
+              },
+              {
+                "any_of" => [
+                  {"past_names" => {"any_satisfy" => {"equal_to_any_of" => ["bar"]}}},
+                  {"forbes_valuations" => {"any_satisfy" => {"lt" => 60_000}}}
+                ]
+              }
+            ]
+          }
+
+          results = search_with_freeform_filter(filter)
+          expect(results).to match_array(ids_of(t2, t3))
+        end
+
+        it "handles not operator" do
+          index_into(
+            graphql,
+            t1 = build(:team, id: "t1", past_names: ["foo"], forbes_valuations: [500_000]),
+            build(:team, id: "t2", past_names: ["foo"], forbes_valuations: [50_000]),
+            build(:team, id: "t3", past_names: ["bar"], forbes_valuations: [500_000])
+          )
+
+          filter = {
+            "all_of" => [
+              {"forbes_valuations" => {"any_satisfy" => {"gt" => 100_000}}},
+              {"not" => {"past_names" => {"any_satisfy" => {"equal_to_any_of" => ["bar"]}}}}
+            ]
+          }
+
+          results = search_with_freeform_filter(filter)
+          expect(results).to match_array(ids_of(t1))
+        end
+
+        it "handles empty filter" do
+          index_into(
+            graphql,
+            t1 = build(:team, id: "t1", forbes_valuations: [500_000]),
+            build(:team, id: "t2", forbes_valuations: [50_000])
+          )
+
+          filter = {
+            "all_of" => [
+              {},
+              {"forbes_valuations" => {"any_satisfy" => {"gt" => 100_000}}}
+            ]
+          }
+
+          results = search_with_freeform_filter(filter)
+          expect(results).to match_array(ids_of(t1))
+        end
+
+        it "handles empty predicates" do
```

Comment:
I think this test and the prior one can be merged into one test--index the data once, and then show two examples of empty filters.  I think of both `{}` and `field: {}` as being equivalent empty predicates--they are just two slightly different examples of the same thing, and as such these two tests aren't showing different facets of behavior.